src/cmd/cmdpkg_create.rb.in

```diff
@@ -0,0 +1,248 @@
+    end
+    #Giving default values
+
+    if !options.key?('init-with-example-system')
+      options['init-with-example-system'] = 0
+    end 
+    if !options.key?('built_type')
+      options['built_type'] = "cmake"
+    end 
+    if !options.key?('export_type')
+      options['export_type'] = "colcon"
+    end 
+    if !options.key?('ign_dependecies')
+      options['ign_dependecies'] = 0
+    end
+    if !options.key?('gazebo_version')
+      options['gazebo_version'] = "fortress"
+    end
+
+    options['command'] = args[0]
+
+    options
+  end # parse()
+
+  def execute(args)
+    options = parse(args)
+
+    # Read the plugin that handles the command.
+    if LIBRARY_NAME[0] == '/'
+      # If the first character is a slash, we'll assume that we've been given an
+      # absolute path to the library. This is only used during test mode.
+      plugin = LIBRARY_NAME
+    else
+      # We're assuming that the library path is relative to the current
+      # location of this script.
+      plugin = File.expand_path(File.join(File.dirname(__FILE__), LIBRARY_NAME))
+    end
+    conf_version = LIBRARY_VERSION
+
+    begin
+      Importer.dlload plugin
+    rescue DLError => e
+      puts "Library error for [#{plugin}]: #{e.to_s}"
+      exit(-1)
+    end
+
+    ##Code for package generation
+    if Dir.exists?(options['name'])
+      puts "ERROR:Folder with this name already exists."
+      exit(-1)
+    end
+
+    #Creates Package folder
+    Dir.mkdir(options['name'])
+
+    #Fetching dependencies versions for common dependencies
+    if options['ign_dependecies']==1
+      dependencies = URI.open('https://raw.githubusercontent.com/ignition-tooling/gazebodistro/master/collection-%s.yaml'% options['gazebo_version']){|f| f.read}
+      dependencies_yaml = YAML.load(dependencies)
+      dependencies_dict={}
+      for dependency in dependencies_yaml.fetch('repositories').keys
+        dependencies_dict[dependency]=dependencies_yaml['repositories'][dependency]['version'].scan( /\d+$/ ).first
+      end
+    end
+
+    resources_dir=__dir__+"/resources"
+
+    #Creates CMakeLists.txt using erb
+    cmakelists_render = ERB.new(File.read(resources_dir+"/CMakeLists.erb"),trim_mode: ">")
+    cmakelists_file = File.new(options['name']+"/CMakeLists.txt", "w")
+    cmakelists_file.puts(cmakelists_render.result(OpenStruct.new(options).instance_eval { binding }))
+    cmakelists_file.close
+    
+    #Creates package.xml for ament dependent
+    if options['built_type']=='ament_cmake'
+      package_xml_render = ERB.new(File.read(resources_dir+"/package_xml.erb"),trim_mode: ">")
+      package_xml_file = File.new(options['name']+"/package.xml", "w")
+      package_xml_file.puts(package_xml_render.result(OpenStruct.new(options).instance_eval { binding }))
+      package_xml_file.close
+    end
+
+    #Creates example resources for advance package
+    if options['init-with-example-system']==1
+      Dir.mkdir(options['name']+"/worlds")
```

Comment:
This bit is hard coded, why not have a separate config? For instance in the templates folder have a file called `template.yaml` inside this we could have somthing like:
```
init-with-example-systems:
  - include_files: ["executables/random_points.cc", ....]
ament_package:
  - include_files: ["file_x", "file_y"...]
some_other_future_config:
  - ... 
```
Then when its an erb file we run it through a struct.

Reply:
It will be pretty much same thing, for new options we will have to add that variable everywhere

Reply:
No it will not. Having a yaml file allows for separation of concern. Having someone hard code these values and expecting them to go and fix them is a nightmare for a maintainer.